src/fs/file.rs

```diff
@@ -336,8 +340,12 @@ impl File {
     /// ```
     ///
     /// [`Ok(n)`]: Ok
-    pub async fn write_at<T: IoBuf>(&self, buf: T, pos: u64) -> crate::BufResult<usize, T> {
-        let op = Op::write_at(&self.fd, buf, pos).unwrap();
+    pub async fn write_at<T: IoBuf>(
+        &self,
+        buf: impl Into<Slice<T>>,
+        pos: u64,
+    ) -> crate::BufResult<usize, Slice<T>> {
```

Comment:
Similarly, the returned slice here could refer to the unwritten portion of the input buffer/slice.

Reply:
Yes. Probably not for this PR unless @Noah-Kennedy chimes in and says this is the right time to make that leap. I suspect it is something he and @carllerche are going to want to think about. The whole idea of a buffer representing filled data or initialized data is something that tokio too is dealing with and my last reading was they weren't happy with their APIs either.